docs/src/pages/components/data-grid/filtering/QuickFilteringGrid.js

```diff
@@ -0,0 +1,119 @@
+import * as React from 'react';
+import IconButton from '@material-ui/core/IconButton';
+import TextField from '@material-ui/core/TextField';
+import {
+  DataGrid,
+  GridToolbarColumnsButton,
+  GridToolbarContainer,
+  GridToolbarDensitySelector,
+  GridToolbarFilterButton,
+} from '@material-ui/data-grid';
+import { useDemoData } from '@material-ui/x-grid-data-generator';
+import ClearIcon from '@material-ui/icons/Clear';
+import SearchIcon from '@material-ui/icons/Search';
+import { createMuiTheme } from '@material-ui/core/styles';
+import { makeStyles } from '@material-ui/styles';
+
+const defaultTheme = createMuiTheme();
+const useStyles = makeStyles(
+  (theme) => ({
+    toolbar: {
+      '&&': {
+        padding: theme.spacing(1, 1, 0),
+      },
+    },
+    search: {
+      boxSizing: 'unset',
+    },
+    searchInputUnderline: {
+      '&:before': {
+        borderBottom: `1px solid ${theme.palette.divider}`,
+      },
+    },
+  }),
+  { defaultTheme },
+);
+
+function QuickSearchToolbar(props) {
+  const classes = useStyles();
+
+  return (
+    <GridToolbarContainer
+      className={classes.toolbar}
+      style={{
+        justifyContent: 'space-between',
+      }}
```

Comment:
root convention + no inline style

```suggestion
      className={classes.root}
```